src/modules/limitOrders/pure/OrderStatusBox/index.tsx

```diff
@@ -55,16 +57,24 @@ const Wrapper = styled.div<{
   }
 `
 
-export type OrderStatusBoxProps = { order: ParsedOrder; widthAuto?: boolean; withWarning?: boolean }
+export type OrderStatusBoxProps = {
+  order: ParsedOrder
+  widthAuto?: boolean
+  withWarning?: boolean
+  clickable?: boolean
+  onClick?: () => void
+}
 
-export function OrderStatusBox({ order, widthAuto, withWarning }: OrderStatusBoxProps) {
+export function OrderStatusBox({ order, widthAuto, withWarning, clickable, onClick }: OrderStatusBoxProps) {
   return (
     <Wrapper
       cancelling={order.isCancelling}
       partiallyFilled={order.partiallyFilled}
       status={order.status}
       widthAuto={widthAuto}
       withWarning={withWarning}
+      clickable={clickable}
```

Comment:
question, why do you add clickable, shouldn't this be derived from `onClick`

`const clickable = !!onClick`

Reply:
Yes that makes sense, thank you for your suggestion.

Reply:
@anxolin I did an update to passing the prop:

`<OrderStatusBox order={order} withWarning={withWarning} clickable={!!onClick} onClick={onClick} />`

`clickable` is only `true` if there is an `onClick` event.

The reason for passing two separate props is because we use `clickable` in the styled component as a passed prop. `onClick` is a function, but the styled component uses `clickable` as a `boolean`. I don't think it's wise to merge this into one single prop here?

Reply:
I made a PR with what i was suggesting https://github.com/cowprotocol/cowswap/pull/2565

The component is easier to use, and less prone to errors. You are not intended to send clickable false, if you pass a clickable callback

Reply:
@anxolin Just checked, and makes sense, thanks!